### DISCUSSION TITLE
hide class input

### DISCUSSION THREAD

**Discussion by vizardkill** [GrapesJS/grapesjs#4723] [Q&A] [answered]
Hello dear Grapes community.

I have a question, I have built a custom block but I would like it not to be possible to inject CSS classes, in the block, it is also possible to hide it, and in the StyleManager it will only show a message that says "This block cannot be customized" or something So?

![image](https://user-images.githubusercontent.com/37307788/201228157-44bd23ee-cab8-4114-991d-07b0ecfdc26e.png)

**Comment by artf** [comment, marked as answer]
You can rely on component selection event for such a case
```js
editor.on('component:selected', (component) => {
      if (component.is('my-component-type')) {
        // hide stuff in style manager
      } else {
        // show stuff in style manager
      }
});
```
